src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -111,11 +112,24 @@
     private transient Project gitlabProject;
     private int projectId = -1;
 
+    /**
+     * The cache of {@link ObjectMetadataAction} instances for each open MR.
+     */
+    @NonNull
+    private transient /*effectively final*/ Map<Integer,ObjectMetadataAction> mergeRequestMetadataCache;
```

Comment:
You could move initialization on the field itself.

otherwise you should add a `readResolve` method see https://github.com/jenkinsci/github-branch-source-plugin/blob/0d65789ce8cc627bb33e76aae2863b8c95b2be95/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java#L509-L514

```suggestion
    private transient /*effectively final*/ Map<Integer,ObjectMetadataAction> mergeRequestMetadataCache = new ConcurrentHashMap<>();
```